internal/config.go

```diff
@@ -0,0 +1,63 @@
+package internal
+
+import (
+	"strings"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// InitConfig intializes the config for the application
+// If no config file is provided with the --config flag
+// it will look for a config in following locations:
+//
+// ${context} being one of the supported components: agent|web|runner
+//
+// Order represents priority
+// /etc/trento/${context}.yaml
+// /usr/etc/trento/${context}.yaml
+// $HOME/${context}.yaml
+func InitConfig(configName string) error {
+	var shouldBreakOnConfigNotFound bool
+
+	cfgFile := viper.GetString("config")
+	if cfgFile != "" {
+		// if a config file has been explicitly provided by --config flag,
+		// then we should break if that file does not exist
+		shouldBreakOnConfigNotFound = true
+
+		// Use config file from the flag.
+		viper.SetConfigFile(cfgFile)
+	} else {
+		// otherwise, if no configuration file was explicitly provided,
+		// we should look for a config in the expected locations
+		// and if no configuration was found in none of the locations, we just passthrough
+		shouldBreakOnConfigNotFound = false
+
+		// Find home directory.
+		home, err := homedir.Dir()
+		cobra.CheckErr(err)
+
+		viper.AddConfigPath("/etc/trento/")
+		viper.AddConfigPath("/usr/etc/trento/")
+		viper.AddConfigPath(home)
+		viper.SetConfigName(configName)
+	}
+
+	SetLogLevel(viper.GetString("log-level"))
+	SetLogFormatter("2006-01-02 15:04:05")
+	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_", ".", "_"))
+	viper.SetEnvPrefix("TRENTO")
+	viper.AutomaticEnv() // read in environment variables that match
+
+	err := viper.ReadInConfig()
+
+	if err != nil {
+		if _, notFoundError := err.(viper.ConfigFileNotFoundError); !notFoundError || shouldBreakOnConfigNotFound {
+			return err
+		}
```

Comment:
I think we have some error conditions not being handled here, e.g. other types of file I/O errors that `viper.ReadInConfig()` might return.